### PR TITLE
Xe/test install script libvirtd

### PR DESCRIPTION
**Pull request by Xe**
- [x] Alpine Edge
- [x] Amazon Linux
- [x] Arch Linux
- [x] Centos 7
- [x] Centos 8
- [ ] Debian Bullseye (https://github.com/tailscale/tailscale/issues/1937)
- [x] Debian Buster
- [x] Debian Sid
- [x] Debian Stretch
- [x] Fedora (34)
- [x] FreeBSD 12
- [x] FreeBSD 13
- [x] OpenSUSE Leap 15.1
- [x] OpenSUSE Leap 15.2
- [x] OpenSUSE Tumbleweed
- [x] Ubuntu 16.04
- [x] Ubuntu 18.04
- [x] Ubuntu 19.10 (https://github.com/tailscale/tailscale/issues/1936)
- [x] Ubuntu 20.04
- [x] Ubuntu 20.10
- [x] Ubuntu 21.04

Testing methodology: Use the [nocloud](https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html) data source for cloud-init and a fleet of cloud-init enabled images. TODO(Xe): `md5sum lufta:~/dl/iso/qcow2` and write a simple cloud-init script to auto-login your instance to Tailscale.

Images were imported like this:

```console
$ sudo zfs create -o mountpoint=none rpool/ts

$ qemu-img info amzn2-kvm-2.0.20210427.0-x86_64.xfs.gpt.qcow2
image: amzn2-kvm-2.0.20210427.0-x86_64.xfs.gpt.qcow2
file format: qcow2
virtual size: 25 GiB (26843545600 bytes)
disk size: 591 MiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    compression type: zlib
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
    extended l2: false

$ sudo zfs create -V 25G rpool/ts/amazon-linux
$ sudo qemu-img convert -O raw amzn2-kvm-2.0.20210427.0-x86_64.xfs.gpt.qcow2 /dev/zvol/rpool/ts/amazon-linux
```

My cloud-init script looked like this:

```yaml
#cloud-config
#vim:syntax=yaml

cloud_config_modules:
 - runcmd

cloud_final_modules:
 - [users-groups, always]
 - [scripts-user, once-per-instance]

users:
  - name: xe
    groups: [ wheel ]
    sudo: [ "ALL=(ALL) NOPASSWD:ALL" ]
    shell: /bin/bash
    ssh-authorized-keys: 
    - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIPg9gYKVglnO2HQodSJt4z4mNrUSUiyJQ7b+J798bwD9 cadey@shachi

write_files:
  - path: /etc/cloud/cloud.cfg.d/80_disable_network_after_firstboot.cfg
    content: |
      # Disable network configuration after first boot
      network:
        config: disabled

runcmd:
 - [ yum, update, gnupg2 ]
 - [ "sh", "-c", "curl https://raw.githubusercontent.com/tailscale/tailscale/Xe/test-install-script-libvirtd/scripts/installer.sh | sh" ]
 - [ systemctl, enable, --now, tailscaled.service ]
 - [ sh, "-c", "sleep 2" ]
 - [ tailscale, up, "--authkey=lolnotpastingthathere" ]
```

If something hecks up, look in the system journal and try to find an error, then fix it by running the install steps by hand such that no human interaction is required. You should also reflash the disk from the VM image to be on the safe side.

```
$ md5sum *
1dc17aefb1f5661d3965317a48d6bc27  amzn2-kvm-2.0.20210427.0-x86_64.xfs.gpt.qcow2
f475de61c5878f4c661481ac8236a4fa  Arch-Linux-x86_64-cloudimg-20210515.22945.qcow2
ef7f109590094e37c54aead73f3cdedc  CentOS-7-x86_64-GenericCloud.qcow2
bf653cc2b5becb29c6cf7c6f7ecaf70f  CentOS-8-GenericCloud-8.3.2011-20201204.2.x86_64.qcow2
bc3186a64e3c0a4c75b71512fd2933ef  debian-9.13.21-20210511-openstack-amd64.qcow2
146723edf45e465a69e120b4ae809aba  debian-10-generic-amd64-20210329-591.qcow2
f53b3055410568969a20bd4fca925ed9  debian-11-generic-amd64-daily-20210515-638.qcow2
b85468f217d229815f3e5f7b385213b0  debian-sid-generic-amd64-daily-20210515-638.qcow2
99abdbe1058f58186648f97a33cefdc0  Fedora-Cloud-Base-34-1.2.x86_64.qcow2
34147b64d78bca01d0d7c3674bf4bf73  openSUSE-Leap-15.1-OpenStack.x86_64.qcow2
343e3d526345556a9d9a0026dd8f4d21  openSUSE-Leap-15.2.x86_64-NoCloud.qcow2
e7924de055adc967064d7b42eb4f4d16  ubuntu-16.04-amd64.qcow2
71c54cd3435a67f979b2bfda4aa62c8f  ubuntu-18.04-amd64.qcow2
56d956a1e382c629f0fc8406b3967aa2  ubuntu-20.04-amd64.qcow2
8930361be88992c7a7f50d4ed8d42c87  ubuntu-20.10-amd64.qcow2
78f7466c54a10dfbb0a7f523c730260c  ubuntu-21.04-amd64.qcow2
```

### CODE REVIEW

**Comment by Xe** [comment]
To run tumbleweed in a VM, you need to use the Tumbleweed JeOS OpenStack Cloud image from here: https://download.opensuse.org/tumbleweed/appliances/

File: scripts/installer.sh [hunk 1]
```diff
@@ -325,45 +333,47 @@ main() {
...
+			$SUDO dnf install -y tailscale
 			$SUDO systemctl enable --now tailscaled
 			set +x
 		;;
 		zypper)
 			set -x
-			$SUDO zypper ar -g -r "https://pkgs.tailscale.com/stable/$OS/$VERSION/tailscale.repo"
-			$SUDO zypper ref
-			$SUDO zypper in tailscale
+      $SUDO rpm --import https://pkgs.tailscale.com/$RELEASE/$OS/$VERSION/repo.gpg
```
Comment (bradfitz) [inline scripts/installer.sh:359]: tabs

**Review by bradfitz — CHANGES_REQUESTED** [review]
Many lines are indented with spaces instead of tabs. Use tabs for consistency.

**Comment by bradfitz** [comment]
Either squash this down to 1 commit, or at least squash away interim WIP fixes like "scripts/installer: forgot a sigil"

File: scripts/installer.sh [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 set -eu
 
+RELEASE="${RELEASE:-stable}"
```
Comment (bradfitz) [inline scripts/installer.sh:11]: Maybe `RELEASE="${TAILSCALE_RELEASE:-stable}"` ?

**Comment by bradfitz** [comment]
LGTM but maybe @danderson has thoughts.

**Comment by bradfitz** [comment]
Closing for now; seems stale.

**Pull request closed without merging**